authnz/src/main/java/io/stargate/auth/AuthenticationService.java

```diff
@@ -15,11 +15,16 @@
  */
 package io.stargate.auth;
 
+import io.stargate.db.Authenticator.SaslNegotiator;
+
 public interface AuthenticationService {
 
   String createToken(String key, String secret) throws UnauthorizedException;
 
   String createToken(String key) throws UnauthorizedException;
 
   StoredCredentials validateToken(String token) throws UnauthorizedException;
+
+  SaslNegotiator getSaslNegotiator(
+      SaslNegotiator wrapped, String tokenUsername, int tokenMaxLength);
```

Comment:
Why are `tokenUsername` and `tokenMaxLength` passed as arguments?  These parameters are implementation details and should be constants in the specific implementing class.

Reply:
done